src/services/usercentrics-service.ts

```diff
@@ -27,36 +30,61 @@ const subscribe = (x: (x: ServiceState) => void): (() => void) => {
     }
 }
 
-const setIsOpen = (isOpen: boolean) => {
+const setIsOpen = (isOpen: boolean, hasInteracted?: boolean) => {
     state = { ...state, isOpen }
+    if (hasInteracted !== undefined) state.hasInteracted = hasInteracted
     update()
 }
-const setIsInitialized = (initialized: boolean) => {
-    state = { ...state, initialized }
+
+const setIsInitialized = (isInitialized: boolean) => {
+    state = { ...state, isInitialized }
     update()
 }
 
 const onEvent = (event: Event) => {
     if (isUCUICMPEvent(event)) {
         switch (event.detail.type) {
-            case UCUICMPEventType.CMP_SHOWN:
+            case UCUICMPEventType.CMP_SHOWN: {
                 setIsOpen(true)
                 break
+            }
+
             case UCUICMPEventType.ACCEPT_ALL:
             case UCUICMPEventType.DENY_ALL:
-            case UCUICMPEventType.SAVE:
-                setIsOpen(false)
+            case UCUICMPEventType.SAVE: {
+                setIsOpen(false, true)
```

Comment:
setIsOpen(false, true) ?

maybe this needs a little bit of attention....

Reply:
I edited this to a more generic `setState({ isOpen: false, hasInteracted: true })`  so it's more clear.